kobo/apps/trash_bin/tests/storage_cleanup/test_attachment_cleanup.py

```diff
@@ -0,0 +1,217 @@
+            '_submitted_by': self.owner.username,
+        }
+        self.asset.deployment.mock_submissions([submission])
+        self.assertEqual(Attachment.objects.all().count(), 1)
+
+        with patch(
+            'kobo.apps.subsequences.api_view.ServiceUsageCalculator.get_usage_balances',  # noqa
+            return_value=updated_balances
+        ):
+            auto_delete_excess_attachments(self.owner.pk)
+
+        # The new attachment should *not* be deleted since user is no longer over limit
+        self.assertEqual(Attachment.objects.all().count(), 1)
+
+    @pytest.mark.skipif(
+        not settings.STRIPE_ENABLED, reason='Requires stripe functionality'
+    )
+    @override_config(AUTO_DELETE_ATTACHMENTS=True)
+    def test_schedule_cleanup_task_only_for_users_exceeding_grace_period(self):
+        """
+        Test that only users over their limit for more than the grace period are
+        scheduled for attachment cleanup
+        """
+        ExceededLimitCounter.objects.create(
+            user=self.owner,
+            limit_type=UsageType.STORAGE_BYTES,
+            days=config.LIMIT_ATTACHMENT_REMOVAL_GRACE_PERIOD + 1
+        )
+
+        with patch(
+            'kobo.apps.trash_bin.tasks.attachment.auto_delete_excess_attachments.delay'
+        ) as mock_task:
+            schedule_auto_attachment_cleanup_for_users()
+            mock_task.assert_called_once_with(self.owner.pk)
+
+    @pytest.mark.skipif(
+        not settings.STRIPE_ENABLED, reason='Requires stripe functionality'
+    )
+    @override_config(AUTO_DELETE_ATTACHMENTS=True)
+    def test_schedule_cleanup_task_skips_users_below_grace_period(self):
+        """
+        Test that users who are over their limit but within the grace period
+        are not scheduled for attachment cleanup
+        """
+        ExceededLimitCounter.objects.create(
+            user=self.owner,
+            limit_type=UsageType.STORAGE_BYTES,
+            days=config.LIMIT_ATTACHMENT_REMOVAL_GRACE_PERIOD - 1
+        )
+
+        with patch(
+            'kobo.apps.trash_bin.tasks.attachment.auto_delete_excess_attachments.delay'
+        ) as mock_task:
+            schedule_auto_attachment_cleanup_for_users()
+            mock_task.assert_not_called()
+
+    def test_auto_delete_excess_attachments_ignores_missing_balance_info(self):
+        """
+        If `ServiceUsageCalculator` returns no info for 'storage_bytes',
+        nothing should be deleted
+        """
+        with patch(
+            'kobo.apps.trash_bin.tasks.attachment.ServiceUsageCalculator.get_usage_balances'  # noqa
+        ) as mock_usage:
+            mock_usage.return_value = {}
+
+            auto_delete_excess_attachments(self.owner.pk)
+            self.assertTrue(Attachment.objects.filter(pk=self.attachment.pk).exists())
+
+    @pytest.mark.skipif(
+        not settings.STRIPE_ENABLED, reason='Requires stripe functionality'
+    )
+    @override_config(AUTO_DELETE_ATTACHMENTS=True)
```

Comment:
What about adding a test for `AUTO_DELETE_ATTACHMENTS=False`  where nothing is called? 
It could be useful.